src/transformers/models/led/modeling_tf_led.py

```diff
@@ -657,7 +672,8 @@ def _chunk(hidden_states, window_overlap):
     def _get_global_attn_indices(is_index_global_attn):
         """ compute global attn indices required throughout forward pass """
         # helper variable
-        num_global_attn_indices = tf.reduce_sum(tf.cast(is_index_global_attn, dtype=tf.dtypes.int32), axis=1)
+        num_global_attn_indices = tf.math.count_nonzero(is_index_global_attn, axis=1)
+        num_global_attn_indices = tf.cast(num_global_attn_indices, dtype=tf.constant(1).dtype)
```

Comment:
This is creating a variable in the current dtype is using that to cast the variable? Isn't there a more efficient way of obtaining the current runtime's dtype?

Reply:
Very good remark! Unfortunately there is no way to detect the `dtype` policy for integers, only for floats with `self._compute_dtype` is allowed inside a layer to detect `float16` or `float32`.

I don't like it neither, I can do an exception here and replace it with usual `tf.int32`. Do you prefer @LysandreJik?

Reply:
If there's no way then it's fine to leave it like that! What happens if you don't cast it?

Reply:
`num_global_attn_indices` if not casted are in `tf.int64` and the integration tests becomes a bit unstable because of this. So I have to keep the cast to "normal" integers.